spec/lib/simplekiq/monitored_job_status_spec.rb

```diff
@@ -0,0 +1,171 @@
+require 'spec_helper'
+
+RSpec.describe MonitoredJobStatus do
+  let(:expiry_time) { (Time.now + expiry).iso8601 }
+  let(:expiry) { 60 }
+  let(:job_id) { SecureRandom.hex }
+  let(:job_status) { MonitoredJobStatus.new(job_id) }
+  let(:params) { { job_id: job_id, total: 100, expiry: expiry } }
+  let(:start_time) { Time.now.iso8601 }
+  before { Timecop.freeze }
+  after { Timecop.return }
+
+  describe '.setup_new_job' do
+    subject { described_class.setup_new_job(params) }
+
+    it 'initializes job status' do
+      subject
+      expect(job_status).not_to be_nil
+    end
+
+    it 'sets job start time' do
+      subject
+      expect(job_status.start_at).to eq start_time
+    end
+
+    it 'sets job status expiry time' do
+      subject
+      expect(job_status.expire_at).to eq expiry_time
+    end
+
+    it 'sets total jobs' do
+      subject
+      expect(job_status.total).to eq 100
+    end
+  end
+
+  describe '.worker_successful!' do
+    subject { described_class.worker_successful!(job_id) }
+    let(:success_count) { 101 }
+
+    context 'when the job status has not been initialized' do
+      subject { described_class.worker_successful!(job_id, success_count) }
+
+      it 'sets successful count' do
+        subject
+        expect(job_status.successful).to eq success_count
+      end
+    end
+
+    context 'when the job status has been initialized' do
+      before do
+        described_class.setup_new_job(params)
+        described_class.worker_successful!(job_id, 101)
+      end
+
+      it 'increments the successful count' do
+        subject
+        expect(job_status.successful).to eq success_count + 1
+      end
+    end
+  end
+
+  describe '.worker_failed!' do
+    subject { described_class.worker_failed!(job_id) }
+    let(:failed_count) { 101 }
```

Comment:
If 101 is supposed to reference the params[:total] + 1
can it be:
```suggestion
    let(:failed_count) { params[:total] + 1 }
```